lib/ansible/modules/database/postgresql/postgresql_ext.py

```diff
@@ -175,15 +194,15 @@ def main():
     try:
         if module.check_mode:
             if state == "present":
-                changed = not ext_exists(cursor, ext)
+                changed = not ext_exists(cursor, ext, ver)
             elif state == "absent":
-                changed = ext_exists(cursor, ext)
+                changed = ext_exists(cursor, ext, ver)
         else:
             if state == "absent":
                 changed = ext_delete(cursor, ext)
```

Comment:
`version` must be checked for deletion operation. Moreover when `check_mode` is enabled, `version` is checked: when `check_mode` is enabled one particular version is tested, when `check_mode` is disabled any version is deleted.